pyodide_build/mkpkg.py

```diff
@@ -71,20 +75,54 @@ def make_package(package):
         yaml.dump(yaml_content, fd, default_flow_style=False)
 
 
+def update_package(package):
+    import yaml
+
+    with open(PACKAGES_ROOT / package / 'meta.yaml', 'r') as fd:
+        yaml_content = yaml.load(fd, Loader=yaml.FullLoader)
+
+    if set(yaml_content.keys()) != set(('package', 'source', 'test')):
+        print(
+            'Only pure-python packages can be updated using this script. '
+            'Aborting.')
+        sys.exit(1)
+
+    sdist_metadata, pypi_metadata = _get_metadata(package)
+    pypi_ver = pypi_metadata['info']['version']
+    local_ver = yaml_content['package']['version']
+    if pypi_ver <= local_ver:
+        print(f'Already up to date. Local: {local_ver} Pypi: {pypi_ver}')
+        sys.exit(0)
+    print(f'Updating {package} from {local_ver} to {pypi_ver}')
+
+    if 'patches' in yaml_content['source']:
+        import warnings
+        warnings.warn(
+            f"Pyodide applies patches to {package}. Update the "
+            "patches (if needed) to avoid build failing.")
+
+    yaml_content['source']['url'] = sdist_metadata['url']
+    yaml_content['source'].pop('md5', None)
+    yaml_content['source']['sha256'] = sdist_metadata['digests']['sha256']
+    yaml_content['package']['version'] = pypi_metadata['info']['version']
```

Comment:
Rather than duplicating this from `make_package`, could we just perform a [dictionary merge](https://gist.github.com/angstwad/bf22d1822c38a92ec0a9) between the data that `make_package` produces and the yaml content loaded from disk?  Probably would require refactoring to create a function that just returns the `yaml_content` variable from `make_package` and using that.  I'm just worried about these code paths diverging overtime (particularly as the package format becomes more complex).

Reply:
My first thought was to do exactly as you suggested. In the end I found this approach more grokable. Not requiring the knowledge of why a recursive merge was needed or why we couldn't just regenerate the entire file was a plus of this solution.